eng/mgmt/automation/generate_data.py

```diff
@@ -54,20 +55,30 @@ def sdk_automation_cadl(config: dict) -> List[dict]:
                 module = match.group(2)
 
             if not module:
-                logging.warning('[Skip] sdk-folder option {} not match format sdk/<service>/<module>'
-                                .format(sdk_folder))
+                logging.warning('[Skip] "emitter-output-dir" not match format '
+                                '{java-sdk-folder}/sdk/{service-directory-name}/<module>')
             else:
                 # cadl
                 succeeded = False
                 pwd = os.getcwd()
                 os.chdir(cadl_dir)
                 try:
+                    # install latest @azure-tools/cadl-java
+                    subprocess.run('npm install @azure-tools/cadl-java', shell=True, check=True)
+
+                    # install dependencies
                     subprocess.run('npm install', shell=True, check=True)
```

Comment:
Alternative to use fixed version. We may needs it because cadl's new version may break our pipeline.
```suggestion
                   # install latest @azure-tools/cadl-java
                    subprocess.run('npm install "@azure-tools/cadl-java@0.2.0"', shell=True, check=True)

                    # install dependencies
                    subprocess.run('npm install "@cadl-lang/compiler@~0.38.5" "@cadl-lang/versioning@0.38.0" "@azure-tools/cadl-autorest@0.24.0" "@cadl-lang/eslint-config-cadl@0.5.0" "@cadl-lang/eslint-plugin@0.38.0" "@cadl-lang/library-linter@0.38.0"', shell=True, check=True)
```